pkg/hubble/relay/observer/server.go

```diff
@@ -276,18 +277,21 @@ func (s *Server) ServerStatus(ctx context.Context, req *observerpb.ServerStatusR
 	g, ctx = errgroup.WithContext(ctx)
 
 	peers := s.peers.List()
+	mu := lock.Mutex{}
 	var numConnectedNodes, numUnavailableNodes uint32
 	var unavailableNodes []string
 	statuses := make(chan *observerpb.ServerStatusResponse, len(peers))
 	for _, p := range peers {
 		if !isAvailable(p.Conn) {
+			mu.Lock()
 			numUnavailableNodes++
 			s.opts.log.WithField("address", p.Address).Infof(
 				"No connection to peer %s, skipping", p.Name,
 			)
 			if len(unavailableNodes) < numUnavailableNodesReportMax {
 				unavailableNodes = append(unavailableNodes, p.Name)
 			}
+			mu.Unlock()
 			continue
 		}
 		numConnectedNodes++
```

Comment:
I think we need to hold `mu` when incrementing this one as well.

Overall, maybe we should only keep track of `numUnavailableNodes` and compute `numConnectedNodes` as `len(peers) - numUnavailableNodes` at the end. Both the math and locking would be a tad simpler, what do you think?